cupy/core/_routines_math.pyx

```diff
@@ -463,6 +469,13 @@ cpdef scan_core(ndarray a, axis, op, dtype=None, ndarray out=None):
         result[...] = a
     if axis is None:
         result = result.ravel()
+        if cupy.cuda.cub_enabled:
+            # result will be None if the scan is not compatible with CUB
+            cub_ops = {scan_op.SCAN_SUM: cub.CUPY_CUB_CUMSUM,
+                       scan_op.SCAN_PROD: cub.cub.CUPY_CUB_CUMPROD}
```

Comment:
For the performance, how about constructing the dictionary outside of this function, or writing with if-branch without the dictionary?

Reply:
Done! thanks